### ISSUE TITLE
Trying to access array offset on value of type int DefaultValueBinder.php in line 57

### ISSUE THREAD

**Issue by rafaeldiasbh** [PHPOffice/PhpSpreadsheet#3321]
This is:

```
- [X] a bug report
- [ ] a feature request
- [ ] **not** a usage question (ask them on https://stackoverflow.com/questions/tagged/phpspreadsheet or https://gitter.im/PHPOffice/PhpSpreadsheet)
```

### What is the expected behavior?
To load the xlsx file

### What is the current behavior?

ErrorException: PHP Error occurred 'Trying to access array offset on value of type int' at file /home/developer/workfolder/erp/modules/est/backend/vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Cell/DefaultValueBinder.php in line 57

### What are the steps to reproduce?
Just try to load the anexed xlsx file, its just 2 lines, created on googlesheets and download as xlsx. Ive noticed the error only happens if i leave the cell format on Automatic.

Please provide a [Minimal, Complete, and Verifiable example](https://stackoverflow.com/help/mcve) of code that exhibits the issue without relying on an external Excel file or a web server:

```php
<?php

require __DIR__ . '/vendor/autoload.php';
use PhpOffice\PhpSpreadsheet\IOFactory;

// Create new Spreadsheet object
$spreadsheet = new \PhpOffice\PhpSpreadsheet\Spreadsheet();

function loadFromBase64($content, $name, $sheetsToLoad = null, Filter $filter = null, $readDataOnly = false){
        $directory = self::createXlsTempDirectory(self::decodeFile($content), $name);
        $xlsReader = self::createXlsReader($directory, $sheetsToLoad, $readDataOnly);
        if($filter != null){
            $xlsReader->setReadFilter($filter);
        }
        $xlsFile  = $xlsReader->load($directory); //returns an object with the content of the file;
        unlink($directory);
        return $xlsFile;
    }

function createXlsTempDirectory($content, $name){
        $fileName = sys_get_temp_dir(). DIRECTORY_SEPARATOR . $name;
        $file = fopen($fileName , 'w+');
        if(!$file){
            throw new \Exception('XLS_READER_OPEN_ERROR');
        }
        fputs($file, $content);
        return $fileName;
    }

function createXlsReader($directory, $sheetsToLoad, $readDataOnly){
        $inputFileType = IOFactory::identify($directory);
        $xlsReader = IOFactory::createReader($inputFileType);
        $xlsReader->setReadDataOnly($readDataOnly);
        $xlsReader->setLoadSheetsOnly($sheetsToLoad);

        return $xlsReader;
    }

```
If this is an issue with reading a specific spreadsheet file, then it may be appropriate to provide a sample file that demonstrates the problem; but please keep it as small as possible, and sanitize any confidential information before uploading.

[Importação dos campos em formato XLS (7) (1).xlsx](https://github.com/PHPOffice/PhpSpreadsheet/files/10512480/Importacao.dos.campos.em.formato.XLS.7.1.xlsx)

### What features do you think are causing the issue

- [X] Reader
- [ ] Writer
- [ ] Styles
- [ ] Data Validations
- [ ] Formula Calculations
- [ ] Charts
- [ ] AutoFilter
- [ ] Form Elements

### Does an issue affect all spreadsheet file formats? If not, which formats are affected?
Tested only with xlsx but i belive it affect them all.

### Which versions of PhpSpreadsheet and PHP are affected?
Tested only on the last release 1.3.1

**Comment by MarkBaker** [comment]
> Tested only on the last release 1.3.1

The last release was `1.27.0`; the `1.3.1` release of PhpSpreadsheet was on Jun 12, 2018; nearly 5 years ago

> Tested only with xlsx but i belive it affect them all.

The Xlsx Reader should never touch the binders at all, because an Xlsx file is typed, so all internal calls to set a cell value in the Reader use explicit methods. A Binder is only used when a datatype is not explicit (e.g. loading from csv files, or from developer calls to set a cell value).

> Just try to load the anexed xlsx file, its just 2 lines, created on googlesheets and download as xlsx. Ive noticed the error only happens if i leave the cell format on Automatic.

The file attached loads without any errors

> ErrorException: PHP Error occurred 'Trying to access array offset on value of type int' at file /home/developer/workfolder/erp/modules/est/backend/vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Cell/DefaultValueBinder.php in line 57

Line 57 in the Default Value Binder is a strict comparison between the value to be stored in the cell and an empty string ([`elseif ($value === ''](https://github.com/PHPOffice/PhpSpreadsheet/blob/1.27.0/src/PhpSpreadsheet/Cell/DefaultValueBinder.php#L57)) {`) which shouldn't ever return an error about array offsets.... and if `$value` was of type int, then the code should never have reached that line, it should have returned a datatype of TYPE_NUMERIC in [line 54](https://github.com/PHPOffice/PhpSpreadsheet/blob/1.27.0/src/PhpSpreadsheet/Cell/DefaultValueBinder.php#L54).

**Comment by rafaeldiasbh** [comment]
Somehow * locked the 1.3.1 version. i ll try to force the newer version and see if it still happens

**Comment by MarkBaker** [comment]
I've just tested your file using version 1.3.1 of PhpSpreadsheet; and it does issue a Deprecation Notice in PHP 7.4, as you indicate (albeit at line 56 rather than line 57), and a Fatal Error in 8.0 and above.

This is because the code is using string offset referencing with square braces, which was deprecated from PHP 7.4 and removed in PHP 8.0.
You are obviously running PHP 8.0 or above.

I would strongly recommend that you update your version of PhpSpreadsheet to work with the more recent versions of PHP

**Comment by rafaeldiasbh** [comment]
Im using PHP 7.4.30 (cli) (built: Jun 27 2022 08:14:10) ( NTS )
But thres some issue with my composer wich isnt installing the phpoffice dependencies automatically, and thats why it probly resolved on 1.3.1

I ll try to fix my composer first and install the last possible version compatible with 7.4
